tests/default_platform_test.py

```diff
@@ -0,0 +1,173 @@
+)
+
+SETUP_OPENSUSE = (
+    'zypper --non-interactive install python > /dev/null 2>&1',
+)
+
+SETUP_ARCH = (
+    'pacman -Syy > /dev/null 2>&1',
+    'pacman -S --noconfirm python > /dev/null 2>&1',
+)
+
+SystemTestCase = namedtuple('SystemTestCase', [
+    'docker_image',
+    'mock_linux_dist',
+    'expected_platform_name',
+    'setup_script',
+])
+
+SYSTEM_TESTCASES = [
+    SystemTestCase(
+        docker_image='ubuntu:lucid',
+        mock_linux_dist=('Ubuntu', '10.04', 'lucid'),
+        expected_platform_name='linux_ubuntu_lucid_x86_64',
+        setup_script=SETUP_DEBIAN,
+    ),
+    SystemTestCase(
+        docker_image='ubuntu:trusty',
+        mock_linux_dist=('Ubuntu', '14.04', 'trusty'),
+        expected_platform_name='linux_ubuntu_trusty_x86_64',
+        setup_script=SETUP_DEBIAN,
+    ),
+    SystemTestCase(
+        docker_image='debian:jessie',
+        mock_linux_dist=('debian', '8.1', ''),
+        expected_platform_name='linux_debian_8_x86_64',
+        setup_script=SETUP_DEBIAN,
+    ),
+    SystemTestCase(
+        docker_image='centos:centos7',
+        mock_linux_dist=('CentOS Linux', '7.1.1503', 'Core'),
+        expected_platform_name='linux_centos_7_x86_64',
+        setup_script=(),
+    ),
+    SystemTestCase(
+        docker_image='fedora:22',
+        mock_linux_dist=('Fedora', '22', 'Twenty Two'),
+        expected_platform_name='linux_fedora_22_x86_64',
+        setup_script=(),
+    ),
+    SystemTestCase(
+        docker_image='rhel7',
+        mock_linux_dist=('Red Hat Enterprise Linux Server', '7.1', 'Maipo'),
+        expected_platform_name='linux_rhel_7_x86_64',
+        setup_script=(),
+    ),
+    SystemTestCase(
+        docker_image='opensuse:13.2',
+        mock_linux_dist=('openSUSE ', '13.2', 'x86_64'),
+        expected_platform_name='linux_opensuse_13_x86_64',
+        setup_script=SETUP_OPENSUSE,
+    ),
+    SystemTestCase(
+        docker_image='base/archlinux',
+        mock_linux_dist=('arch', '', ''),
+        expected_platform_name='linux_arch_unknown_x86_64',
+        setup_script=SETUP_ARCH,
+    ),
+]
+
+
+@pytest.mark.parametrize('case', SYSTEM_TESTCASES)
+@mock.patch('pip_custom_platform.util.platform')
+def test_platform_linux(mock_platform, case):
+    mock_platform.system.return_value = 'Linux'
+    mock_platform.machine.return_value = 'x86_64'
+    mock_platform.linux_distribution.return_value = case.mock_linux_dist
+    assert default_platform_name() == case.expected_platform_name
+
+
+@mock.patch('pip_custom_platform.util.platform')
+@mock.patch('distutils.util.get_platform', return_value='windows')
+def test_platform_notlinux(mock_get_platform, mock_platform):
+    mock_platform.system.return_value = 'Windows'
+    assert default_platform_name() == 'windows'
+    assert mock_get_platform.called
+
+
+@pytest.mark.skipif(
+    'not config.getvalue("docker")',
+    reason="Requires --docker",
+)
+class TestDistributionNameDockerIntegration(object):  # pragma: no cover
```

Comment:
I believe the only way to be sure this stuff works is with the integration tests.
As long as we don't run them by default, it's fine I think.